### PR TITLE
Fixed #27699 -- Added negative timedelta support to parse_duration().

### PR DESCRIPTION
**Pull request by jinankjain**

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
Meanwhile, someone else sent https://github.com/django/django/pull/7817, any thoughts?

File: django/utils/dateparse.py [hunk 1]
```diff
@@ -30,9 +30,9 @@
 standard_duration_re = re.compile(
     r'^'
     r'(?:(?P<days>-?\d+) (days?, )?)?'
-    r'((?:(?P<hours>\d+):)(?=\d+:\d+))?'
-    r'(?:(?P<minutes>\d+):)?'
-    r'(?P<seconds>\d+)'
+    r'((?:(?P<hours>-?\d+):)(?=\d+:\d+))?'
+    r'(?:(?P<minutes>-?\d+):)?'
+    r'(?P<seconds>-?\d+)'
```
Comment (timgraham) [inline django/utils/dateparse.py]: Tests are passing with only this line modified. Are the other changes needed? If so, add tests for them.

Comment (jinankjain) [inline django/utils/dateparse.py]: @timgraham I will update the PR with some more tests

File: django/utils/dateparse.py [hunk 2]
```diff
@@ -125,5 +125,7 @@ def parse_duration(value):
         sign = -1 if kw.pop('sign', '+') == '-' else 1
         if kw.get('microseconds'):
             kw['microseconds'] = kw['microseconds'].ljust(6, '0')
+        if kw.get('seconds') and kw.get('microseconds') and kw['seconds'].startswith('-'):
+            kw['microseconds'] = '-'.__add__(kw['microseconds'])
```
Comment (Stranger6667) [inline django/utils/dateparse.py]: Why not `'-' + kw['microseconds']` ?

**Comment by jinankjain** [comment]
@Stranger6667 agreed 👍  Updated the PR

**Pull request merged**